### PR TITLE
updated async and custom output processor testcase for metamodel

### PR DESCRIPTION
**Pull request by sachintendulkar576123**
## Proposed changes

This PR :

Added the custom output processor testcase for metamodel.
Added async testcase for metamodel

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: tests/langchain_custom_output_processor_test.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            span_processors=[
+                BatchSpanProcessor(HttpSpanExporter("https://localhost:3000/api/v1/traces"))
+            ],
+            wrapper_methods=[
+                WrapperMethod(
+                    package="langchain_core.retrievers",
+                    object_name="BaseRetriever",
+                    method="invoke",
+                    wrapper=task_wrapper,
+                    output_processor=["entities.json"]
```
Comment (oi-raanne) [inline tests/langchain_custom_output_processor_test.py:142]: where is this entities.json loaded from ?

File: src/monocle_apptrace/wrap_common.py [hunk 2]
```diff
@@ -337,23 +348,24 @@ def update_workflow_type(to_wrap, span: Span):
...
+
 def update_span_with_context_output(to_wrap, return_value, span: Span):
     package_name: str = to_wrap.get('package')
     output_arg_text = ""
     if "langchain_core.retrievers" in package_name:
         output_arg_text += " ".join([doc.page_content for doc in return_value if hasattr(doc, 'page_content')])
         if len(output_arg_text) > 100:
             output_arg_text = output_arg_text[:100] + "..."
-    if "llama_index.core.indices.base_retriever" in package_name and len(return_value) > 0:
+    if "llama_index.core.indices.base_retriever" in package_name:
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:368]: Why is the change removed

File: tests/langchain_custom_output_processor_test.py (hunk 1 above)
Comment (sachintendulkar576123) [reply tests/langchain_custom_output_processor_test.py:142]: added entities.json from test folder

File: src/monocle_apptrace/wrap_common.py [hunk 3]
```diff
@@ -190,11 +191,18 @@ async def allm_wrapper(tracer, to_wrap, wrapped, instance, args, kwargs):
     else:
         name = f"langchain.task.{instance.__class__.__name__}"
     with tracer.start_as_current_span(name) as span:
-        update_llm_endpoint(curr_span=span, instance=instance)
+        if 'haystack.components.retrievers' in to_wrap['package'] and 'haystack.retriever' in span.name:
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:194]: Is this code new or just showing new in diff?

File: src/monocle_apptrace/wrap_common.py [hunk 4]
```diff
@@ -93,64 +94,62 @@ def task_wrapper(tracer: Tracer, to_wrap, wrapped, instance, args, kwargs):
...
-            update_span_with_prompt_input(to_wrap=to_wrap, wrapped_args=args, span=span)
-            update_span_with_infra_name(span, INFRA_SERVICE_KEY)
-
-        update_span_with_context_input(to_wrap=to_wrap, wrapped_args=args, span=span)
-    except:
-        logger.exception("exception in pre_task_processing")
-    
+    if is_root_span(span):
+        update_span_with_prompt_input(to_wrap=to_wrap, wrapped_args=args, span=span)
+
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:148]: Why this change?

**Comment by kshitiz-okahu** [comment]
Looks like this PR reverts changes made in previous PR, can you check that?

File: src/monocle_apptrace/wrap_common.py (hunk 3 above)
Comment (sachintendulkar576123) [reply src/monocle_apptrace/wrap_common.py:194]: no this not new code just same same as llm_wrapper but  for allm_wrapper

**Review by oi-raanne — APPROVED** [review]
